lib/trino-collect/src/main/java/io/trino/collect/cache/NonLoadableCache.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.collect.cache;
+
+import com.google.common.cache.Cache;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * A {@link Cache} that does not support loading. So it does not expose issues with eviction.
+ */
+public interface NonLoadableCache<K, V>
```

Comment:
The cache eviction problem isn't limited to `get(K key, Callable<? extends V> loader)` method.
Even if I call `asMap().putIfAbsent` etc i insert into a cache a value that's "fresh" from the calling thread perspective, but can already stale from some other thread perspective.
Thus this safety is an illusion, the `NonLoadableCache` isn't any more safe than any other Guava cache. It's just less convenient to use.